java/dagger/internal/codegen/MapBindingExpression.java

```diff
@@ -99,8 +99,7 @@ Expression getDependencyExpression(ClassName requestingClass) {
             .add("$T.", isImmutableMapAvailable ? ImmutableMap.class : MapBuilder.class)
             .add(maybeTypeParameters(requestingClass));
         if (isImmutableMapAvailable) {
-          // TODO(ronshapiro): builderWithExpectedSize
-          instantiation.add("builder()");
```

Comment:
`builderWithExpectedSize` was added in a newer version of guava - maybe we should have a check similar to the `isImmutableMapAvailable` that checks if `builderWithExpectedSize` exists? 

Something like this (but cleaned up a bit).
```java
isImmutableMapAvailable() && methodsIn(elements.getTypeElement(ImmutableMap.class)).stream().anyMatch(method -> method.getSimpleName().contentEquals("builderWithExpectedSize")
```

WDYT @michaelxinsun @netdpb

Reply:
Thank you @ronshapiro for your reviews!  I like the idea of checking whether "builderWithExpectedSize" is available -- it's not only a "small comment", but also seems absolutely necessary.
Just added "isImmutable{Set|Map}BuilderWithExpectedSizeAvailable" which aligns with "isImmutable{Set|Map}Available".
Maybe @netdpb also has something to say?  Thank you!

Reply:
@ronshapiro @netdpb I'd also like to add a unit test that uses a pre-23.1 Immutable{Set|Map} that does NOT have .builderWithExpectedSize(...), and verify it generates .builder() instead.  Do you have a similar testing infrastructure piece that I can use?  Otherwise I may consider removing this method manually from a mock or something?  Thank you!